### DISCUSSION TITLE
uno-check cant see The framework Microsoft.NETCore.App,even it has installed

### DISCUSSION THREAD

**Discussion by prehonor** [unoplatform/uno#8962] [Q&A]
uno-check --pre
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '6.0.0' (x64) was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=nixos.21.11-x64

dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     nixos
 OS Version:  21.11
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /nix/store/mnmygpa1yiphv6kdgiyjrbf5302p27gj-dotnet-sdk-6.0.100/sdk/6.0.100/

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  6.0.100 [/nix/store/mnmygpa1yiphv6kdgiyjrbf5302p27gj-dotnet-sdk-6.0.100/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [/nix/store/mnmygpa1yiphv6kdgiyjrbf5302p27gj-dotnet-sdk-6.0.100/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/nix/store/mnmygpa1yiphv6kdgiyjrbf5302p27gj-dotnet-sdk-6.0.100/shared/Microsoft.NETCore.App]

**Comment by jeromelaban** [comment]
Thanks for the report. This is an interesting issue.

I wonder, did you by any chance, install .NET in the same terminal for which you ran uno-check? Does it do the same error with another terminal?

**Reply by prehonor** [reply]
Yes,it does.
You mean I should  install dotnet framework manually instead of do  by package manager of Linux ， can I set some  environment variables for uno-check,so it can detect dotnet frameworkd path？

**Reply by jeromelaban** [reply]
Not necessarily, How did you install the runtime?

**Reply by prehonor** [reply]
By package manager，run command “nix-shell -iA dotnet-sdk_6” in NixOs,the nixos is special Linux distribution,Users maybe need set env  var such as“ LD_LIBRARY_PATH”，“PATH ” ， because it has no env var in a terminal

**Reply by jeromelaban** [reply]
I see. I don't know about the details of this specific behavior. NixOS is not part of the [list of supported distros](https://docs.microsoft.com/en-us/dotnet/core/install/linux), so I don't know what would be missing.

In general, the `DOTNET_ROOT` variable is used to determine the location of the runtime/sdk, maybe this is what's missing here. 

You could also try installing manually with [these instructions](https://docs.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install).

 In all cases, this is not something that uno-check does address at this point, as it needs .NET itself to run (for size reasons mainly). We could change uno.check to be self-contained if it becomes a problem.

**Reply by prehonor** [reply]
> I see. I don't know about the details of this specific behavior. NixOS is not part of the [list of supported distros](https://docs.microsoft.com/en-us/dotnet/core/install/linux), so I don't know what would be missing.
> 
> In general, the `DOTNET_ROOT` variable is used to determine the location of the runtime/sdk, maybe this is what's missing here.
> 
> You could also try installing manually with [these instructions](https://docs.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#scripted-install).
> 
> In all cases, this is not something that uno-check does address at this point, as it needs .NET itself to run (for size reasons mainly). We could change uno.check to be self-contained if it becomes a problem.

I make mistakes about Nixos shell env setting, after corrected it and setting DOTNET_ROOT , I sucessed, Thank you very much.
